"echo -n" is not POSIX compliant but printf is.
See "man 1p echo" section "APPLICATION USAGE" or visit
https://www.in-ulm.de/%7Emascheck/various/echo+printf/

--- links-2.17/intl/gen-intl
+++ links-2.17/intl/gen-intl
@@ -10,7 +10,7 @@
 n=0
 echo '/* Automatically generated by gen-intl */'
 for i in $languages; do
-	echo -n $i" " 1>&2
+	printf '%s ' "$i" 1>&2
 	if test `cat $i.lng|wc -l` != `cat english.lng|wc -l`; then
 		echo 1>&2
 		echo 1>&2
--- links-2.17/intl/synclang
+++ links-2.17/intl/synclang
@@ -6,7 +6,7 @@
 if [ -n "$languages" ]; then
 	for i in $languages; do
 		if [ "$i" != english ]; then
-			echo -n $i" " 1>&2
+			printf '%s ' "$i" 1>&2
 			awk -f synclang.awk <$i.lng | sed 's///g' >_sync.tmp
 			mv $i.lng ../intl-bak/$i.lng.bak
 			mv _sync.tmp $i.lng
